### DISCUSSION TITLE
CCU Systemvariablen werden abgeschnitten

### DISCUSSION THREAD

**Discussion by drsamasrd** [SukramJ/aiohomematic#1417] [Q&A] [answered]
Hallo, 

ich stelle fest, dass Systemvariablen, die von der CCU ausgelesen werden, bei 255 Zeichen abgeschnitten werden. Gibt es eine Konfigurationseinstellung in der hahomematic Integration, damit auch längere Inhalte gelesen werden können.

Danke vorab für Kommentare.

Gruß

   Stefan

**Comment by SukramJ** [comment, marked as answer]
Nein, das ist nicht möglich, da die HA-StateMachine nicht mehr als 255 Zeichen speichert.
Das steht auch so in der Fehlermeldung im HA-Protokoll.

**Reply by drsamasrd** [reply]
Meine Idee war auch, dass ich die langen Nachrichten in einen "Helper" schreibe, einen "input_text.*", der länger sein kann. 

Wenn das aber schon beim Abholen der Sysvars geblockt wird, dann geht die Info natürlich verloren. 

Any other idea?

**Reply by SukramJ** [reply]
Und hast du das schon mal ausprobiert in einem `input_helper` mehr als 255 Zeichen zu speichern?

**Reply by drsamasrd** [reply]
Ich hatte einen mit 1000 Max definiert aber das täuscht. Die Doku sagt: 

max Integer (Optional, default: 100)
Maximum length for the text value. 255 is the maximum number of characters allowed in an entity state.

Gibt es keinen Helper mit dem man einen String speichern kann der größer ist, hmm?

**Reply by SukramJ** [reply]
> Nein, das ist nicht möglich, da die HA-StateMachine nicht mehr als 255 Zeichen speichert.

Ich denke nicht, da alles in der StateMachine gespeichert wird.
